app/controllers/candidate_interface/continuous_applications/course_choices/course_site_controller.rb

```diff
@@ -0,0 +1,26 @@
+module CandidateInterface
+  module ContinuousApplications
+    module CourseChoices
+      class CourseSiteController < BaseController
+      private
+
+        def step_params
```

Comment:
This method returns either `ActionController::Parameters` or a `Hash`. It would avoid potential issues later on if the return type was consistent. Probably AC::P as the method name says `_params`

`ActionController::Parameters.new(provider_params)` ?